File: design-patterns/go/factory/transport-method/bus.go
// bus.go

package main

import "fmt"

type Bus struct {
}

func NewBus() (bus *Bus) {
	bus = &Bus{}
	return
}

func (bus *Bus) Repair() {
	fmt.Println("Bus Repair")
}

func (bus *Bus) Start() {
	fmt.Println("Bus started")
}

func (bus *Bus) Stop() {
	fmt.Println("Bus Stopped")
}

File: design-patterns/php/factory/transport-method/Bus.php
<?php
// Bus.php

namespace Factory\TransportMethod;

class Bus implements Transport {
    public function start() {
        echo "Bus started\n";
    }

    public function stop() {
        echo "Bus Stopped\n";
    }

    public function repair() {
        echo "Bus Repair\n";
    }
}